go.mod

```diff
@@ -1,6 +1,9 @@
 module k8s.io/kube-openapi
 
-go 1.21.3
+// Keep this as X.Y (instead of X.Y.Z, which Go 1.21 wants, but Go 1.20 does
+// not support) until we think it is reasonable for consumers to be expected to
+// use Go 1.21 or higher.
+go 1.21
```

Comment:
go 1.21 is really hardnosed about this, and propagates up go versions from dependencies' go directives. 

That means if gengo/v2 says it requires go1.21.x, go 1.21 will refuse to build this module unless it also says it requires go1.21.x

if we ever have to bump to HEAD kube-openapi versions in k/k release branches, go1.21 will refuse to build *those* unless we also bump the release branch k/k go.mod files up to go1.21, which we can't currently do for compat reasons (this is hopefully getting ironed out by go1.23 in https://github.com/golang/go/issues/65573)

If possible, we should leave kube-openapi (and by extension, gengo/v2, I guess) declaring a minimum go version of the oldest in-support k/k branch. That's go1.20 currently. Did anything in gengo/v2 or kube-openapi require anything from go1.21 or go1.22?

Reply:
It's PROBABLY ok to use 1.20, but hard to test.  If I revert go.mod to say "1.20"

```
$ head go.mod 
module k8s.io/kube-openapi

// Keep this as X.Y (instead of X.Y.Z, which Go 1.21 wants, but Go 1.20 does
// not support) until we think it is reasonable for consumers to be expected to
// use Go 1.21 or higher.
go 1.20

require (
	github.com/NYTimes/gziphandler v0.0.0-20170623195520-56545f4a5d46
	github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a

$ go build ./cmd/openapi-gen/
go: updates to go.mod needed; to update it:
	go mod tidy
```

Reply:
to revert this to go1.20 you also have to revert gengo/v2 to go1.20 ... it's viral